pkg/importer/snapshots_importer.go

```diff
@@ -0,0 +1,109 @@
+package importer
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/wavesplatform/gowaves/pkg/proto"
+)
+
+type SnapshotsImporter struct {
+	scheme proto.Scheme
+	st     State
+
+	br  *blocksReader
+	sr  *snapshotsReader
+	reg *speedRegulator
+
+	h uint64
+}
+
+func NewSnapshotsImporter(scheme proto.Scheme, st State, blocksPath, snapshotsPath string) (*SnapshotsImporter, error) {
+	br, err := newBlocksReader(blocksPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create snapshots importer: %w", err)
+	}
+	sr, err := newSnapshotsReader(scheme, snapshotsPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create snapshots importer: %w", err)
+	}
+	return &SnapshotsImporter{scheme: scheme, st: st, br: br, sr: sr, reg: newSpeedRegulator()}, nil
+}
+
+func (imp *SnapshotsImporter) SkipToHeight(ctx context.Context, height proto.Height) error {
+	imp.h = uint64(1)
+	if height < imp.h {
+		return fmt.Errorf("invalid initial height: %d", height)
+	}
+	for {
+		if ctx.Err() != nil {
+			return ctx.Err()
+		}
+		if imp.h == height {
+			return nil
+		}
```

Comment:
It is not necessary to check these conditions, because we increment `imp.h` until it is not equal to `height`. So `height` always greater then `imp.h`